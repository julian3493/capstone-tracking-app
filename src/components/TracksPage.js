import React, { useEffect } from 'react';
import axios from 'axios';
import { useDispatch } from 'react-redux';
import Logout from './Logout';
import Navigation from './Navigation';
import { userTracks } from '../redux/actions/actions';
import Tracks from './Tracks';
import TopBar from './TopBar';
import '../styles/TracksPage.css';

const TracksPage = () => {
  const dispatch = useDispatch();

  const fetchData = async () => {
    await axios.get('https://julian-comsumption-tracker-api.herokuapp.com/tracks',
      { withCredentials: true }).then((response) => {
      if (response.data.registers) {
        const tracks = response.data.registers;
        dispatch(userTracks(tracks));
      }
    }).catch((err) => {
      throw (err);
    });
  };

  useEffect(() => {
    fetchData();
  }, []);

  return (
    <div className="screen">
      {TopBar({ title: 'Tracks' })}
      <Tracks />
      <Logout />
      <Navigation />
    </div>
  );
};

export default TracksPage;
