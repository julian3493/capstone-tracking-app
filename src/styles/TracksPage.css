.screen {
  display: flex;
  flex-direction: column;
  height: 100vh;
}

.track-rows {
  margin: 40px 35px;
  padding: 20px 0;
  background-color: #f3f4f6;
}

.track-row {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  margin-bottom: 5px;
  padding-left: 20px;
  padding-bottom: 10px;
  border-bottom: 1px solid white;
}
