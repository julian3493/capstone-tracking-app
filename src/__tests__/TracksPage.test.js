import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import TracksPage from '../components/TracksPage';
import store from '../redux/store';

it('it tests the render TracksPage correctly', () => {
  const tracksPage = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <TracksPage />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tracksPage).toMatchSnapshot();
});

describe('render correctly the topbar component', () => {
  it('renders the topbar', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <TracksPage />
        </BrowserRouter>
      </Provider>,
    );
    const topbar = document.querySelector('.top');
    expect(topbar).toBeDefined();
  });

  it('renders the page title', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <TracksPage />
        </BrowserRouter>
      </Provider>,
    );
    const topbar = document.querySelector('.top');
    expect(topbar.firstChild.innerHTML).toEqual('Tracks');
  });
});

describe('render correctly the TracksPage content', () => {
  it('should display a tracks container', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <TracksPage />
        </BrowserRouter>
      </Provider>,
    );
    const container = document.querySelector('.track-rows');
    expect(container).toBeDefined();
  });
});

it('should render a logout button', () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <TracksPage />
      </BrowserRouter>
    </Provider>,
  );
  const logoutBtn = document.querySelector('.logoutBtn');
  expect(logoutBtn).toBeDefined();
});

it('should render a navigation component', () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <TracksPage />
      </BrowserRouter>
    </Provider>,
  );
  const nav = document.querySelector('.nav-div');
  expect(nav).toBeDefined();
});
