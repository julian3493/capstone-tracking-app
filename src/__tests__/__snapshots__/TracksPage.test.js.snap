// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it tests the render TracksPage correctly 1`] = `
<div
  className="screen"
>
  <div
    className="top"
  >
    <p
      className="top-title"
    >
      Tracks
    </p>
  </div>
  <div
    className="track-rows"
  >
    <p>
       
    </p>
  </div>
  <button
    className="logoutBtn"
    onClick={[Function]}
    type="button"
  >
    Logout
  </button>
  <div
    className="nav-div"
  >
    <a
      className="nav-link"
      href="/dashboard"
      onClick={[Function]}
    >
      Dashboard
    </a>
    <a
      className="nav-link"
      href="/tracks"
      onClick={[Function]}
    >
      Tracks
    </a>
    <a
      className="nav-link"
      href="/statistics"
      onClick={[Function]}
    >
      Statistics
    </a>
    <a
      className="nav-link"
      href="/more"
      onClick={[Function]}
    >
      More
    </a>
  </div>
</div>
`;
